lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -1237,9 +1238,9 @@ def certificate_issued_for_csr(self, app_name: str, csr: str) -> bool:
             bool: True/False depending on whether a certificate has been issued for the given CSR.
         """
         issued_certificates_per_csr = self.get_issued_certificates()[app_name]
-        for request, cert in issued_certificates_per_csr.items():
-            if request == csr:
-                return csr_matches_certificate(csr, cert)
+        for request_pair in issued_certificates_per_csr:
+            if "csr" in request_pair:
+                return csr_matches_certificate(csr, request_pair["certificate"])
```

Comment:
Are we checking if `csr` was found among the issued ones?

Reply:
I get what you mean now. Fixed.